src/main/java/org/opensearch/knn/index/codec/KNN91Codec/KNN91Codec.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.knn.index.codec.KNN91Codec;
+
+import org.apache.lucene.codecs.Codec;
+import org.apache.lucene.codecs.CompoundFormat;
+import org.apache.lucene.codecs.DocValuesFormat;
+import org.apache.lucene.codecs.FilterCodec;
+import org.apache.lucene.codecs.lucene91.Lucene91Codec;
+import org.opensearch.knn.index.codec.KNNDocFormatFacade;
+import org.opensearch.knn.index.codec.KNNDocFormatFactory;
+
+/**
+ * Extends the Codec to support a new file format for KNN index
+ * based on the mappings.
+ *
+ */
+public final class KNN91Codec extends FilterCodec {
+
+    public static final String KNN_91 = "KNN91Codec";
+    private KNNDocFormatFacade docFormatFacade;
+
+    /**
+     * No arg constructor that uses Lucene91 as the delegate
+     */
+    public KNN91Codec() {
+        this(new Lucene91Codec());
```

Comment:
Is it too much for the KNNDocFormatFactory to also have a `createKNN91DefaultDelegate()` static method? Or `createDefaultDelegate(string version)`?

Reply:
Let me think about that, my concern is that construction of delegate doesn't sound like a function of DocFormat factory. Maybe it's more suitable for a CodecFactory, maybe some inner factory class, like CodecFactory.DelegateFactory. Overall I would love to pull it out of the Codec class by we must have default constructor to be compliant with SPI spec